soundata/datasets/singapura.py

```diff
@@ -0,0 +1,325 @@
+            metadata=metadata,
+        )
+
+        self.audio_path = self.get_path("audio")
+        self.annotation_path = self.get_path("annotation")
+
+    @core.cached_property
+    def annotation(self):
+        """
+        The clip's event annotations
+        Returns:
+            * annotations.Events - sound events with start time, end time, label and confidence
+        """
+        return load_annotation(self.annotation_path)
+
+    @property
+    def audio(self):
+        """
+        The clip's audio
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """
+        Jams: the track's data in jams format
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path, events=self.annotation
+        )
+
+
+@io.coerce_to_string_io
+def load_annotation(fhandle):
+    """
+    Load an annotation file.
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to an annotation file
+
+    Returns:
+        * annotations.Events - sound events with start time, end time, label and confidence
+    """
+
+    df = pd.read_csv(fhandle)
+    intervals = df[["onset", "offset"]].values
+    label = df["event_label"].tolist()
+
+    annotation_data = annotations.Events(
+        intervals=intervals,
+        intervals_unit="seconds",
+        labels=label,
+        labels_unit="open",
+        confidence=np.array([np.nan for _ in label]),
+    )
+
+    return annotation_data
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle):
+    """
+    Load a Example audio file.
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to an audio file
+
+    Returns:
+        * np.ndarray - the audio signal at 44.1 kHz
+    """
+    data, _ = librosa.load(fhandle, sr=44100, mono=False)
+    return data
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    SINGA:PURA v1.0 dataset
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            name="singapura",
+            clip_class=Clip,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            download_info=DOWNLOAD_INFO,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
+
+    @core.copy_docs(load_annotation)
+    def load_annotation(self, *args, **kwargs):
+        return load_annotation(*args, **kwargs)
+
+    @core.cached_property
+    def _metadata(self):
+        metadata_path = os.path.join(self.data_home, "labelled_metadata_public.csv")
+
+        df = pd.read_csv(metadata_path)
+        df["filename"] = df["filename"].apply(lambda x: x.replace(".flac", ""))
+        df = df.set_index("filename")
+
+        metadata = df.to_dict(orient="index")
```

Comment:
The dataset description says that the dataset has spatiotemporal data right? Could you explain a bit more about what data exactly? It might make sense to add it to the `Clip` as well?

Reply:
We have the timestamp, day of the week, and rough location of the sensor included. I have added these to `Clip` in the new commit as well.